lib/ruby_inspector.rb

```diff
@@ -7,30 +6,55 @@
 module RubyInspector
   DELIMITER = "\0"
   class << self
-    def enable(app_name, description = '')
-      connect
-      send_init_info(app_name, description)
+    def enable(app_name, description)
+      @app_name = app_name
+      @description = description
+
+      begin
+        send_init_info
+      rescue Errno::ECONNREFUSED => e
+        puts "[RubyInspector] Unable to send initialization info during setup"
+      end
 
       ::Nsa::NetHttpTracker.on_request do |net_http_request_tracker|
         DevToolsRequestTracker.new(net_http_request_tracker)
       end
     end
 
+    def disable
+      @socket = nil
+      @initialized = nil
+    end
+
     def send_info(data)
-      socket.puts(
-        ::JSON.generate(data) + DELIMITER
-      )
+      begin
+        send_init_info unless initialized?
+        send_socket_msg(data)
+      rescue Errno::ECONNREFUSED => e
+        puts "[RubyInspector] Unable to send data: #{data}"
+      end
     end
 
     private
 
-    attr_accessor :socket
-    def connect
-      @socket = TCPSocket.new('localhost', 8124)
+    attr_accessor :app_name, :description
+
+    def initialized?
+      @initialized
+    end
+
+    def socket
+      @socket ||= TCPSocket.new('localhost', 8124)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.